tests/cases/compiler/noImplicitAnyDestructuringVarDeclaration2.ts

```diff
@@ -0,0 +1,12 @@
+// @noimplicitany: true
+let [a, b, c] = [1, 2, 3]; // no error
+let [a1 = 10, b1 = 10, c1 = 10] = [1, 2, 3]; // no error
+let [a2 = undefined, b2 = undefined, c2 = undefined] = [1, 2, 3]; // no error

```

Comment:
Not really. It is different than that because destructuring(here a2) gets the typing through rhs (here as number because of rhs assignment of [1,2,3])